### ISSUE TITLE
[Accessibility]MAS 1.4.11: The color contrast of the focus dot line on each selection items for LitView/CheckedListBox is below 3:1

### ISSUE THREAD

**Issue by Vino-Wang** [dotnet/winforms#3059]
**.NET Core Version:**
.NET SDK 5.0.100-PREVIEW.4.20213.26 + Latest VS2019

**Have you experienced this same bug with .NET Framework?:**
Yes

**Problem description:**
On navigating to each row using arrow keys (down/up) the focus has a color contrast below 3:1. The ratio is 2.4:1. Low vision users will face difficulty in knowing that there is focus on the rows.
ListView
![ListView](https://user-images.githubusercontent.com/18568603/79197388-389f1300-7e64-11ea-903d-dce0d0838d25.png)

CheckedListBox![CheckedListBox](https://user-images.githubusercontent.com/18568603/79197232-f83f9500-7e63-11ea-8817-17d52ac4fbd6.png)

**Expected behavior:**
When the tab navigation is moved to each row the focus on them should be at a ratio of 3:1 or greater than 3:1.

**Minimal repro:**
1.	Open the attached project, build and run it.
[Non_Text_Contrast_for_focus_dotline.zip](https://github.com/dotnet/winforms/files/4473732/Non_Text_Contrast_for_focus_dotline.zip)
2.	Select one row of ListView or one item of checkedBoxList.
3.	Launch the Accessibility Insights for windows tool, and open the Color Contrast analyzer.
4.	Select the color of the focus dot line.

**Comment by Jessie-Zhang01** [comment]
#3322 tracks the overall support.

**Comment by SergeySmirnov-Akvelon** [comment]
Hi, @merriemcgaw. Do we really need to change the color of the focus rectangle? In a normal situation, this rectangle is so small that even a person without vision problems is hard to consider its color by default. If the color should be changed, then what color should we use? I think this color should be at a ratio of 3:1 or greater than 3:1. for blue and white colors at once

**Comment by merriemcgaw** [comment]
Great questions @SergeySmirnov-Akvelon , I'm going to take a look at the requirements and talk with @guybark about the requirements around contrast of the CheckedListBox.

**Comment by merriemcgaw** [comment]
After discussion with our Accessibility SME we need to do something here to indicate that the item is selected. @SergeySmirnov-Akvelon - is this something that we pick up from CommonControls or is it something that we would be able to reasonably adjust on our end. If we can't make the adjustment, then we can file a bug against Windows to help us out here.

**Comment by SergeySmirnov-Akvelon** [comment]
Hi @merriemcgaw. I have investigated this issue and I have several points.

**CheckedListBox**
When drawing the focusing rectangle, we use a special logic that uses alternation of two colors: black (which is taken by default and contrasts with white and blue) and orange (which is obtained by inverting the blue background color). I agree with the CTI team that perhaps the resulting orange color is not ideal in this case. Therefore, I have several questions:
1. Should we add new logic where instead of inverted color we will use predefined?
2. If so, what color should it be and with what colors (white, blue or black) should it contrast?

**ListBox**
Unfortunately, we do not have direct access to the drawing process of the `ListView`. I tried whether it is possible to add a forced drawing of an additional rectangle, as we did for the buttons in the `PropertyGrid`, but I ran into several problems:
1. Our logic competes with the standard drawing of `ListView` and is superimposed on top of our rectangle.
2. We have problems for calculating the rectangle for an element in `Tile` mode, since the rectangle obtained by standard methods does not coincide with the highlighted blue zone.

I've tried implementing a workaround where the `OwnerDraw` option is enabled and the user is allowed to draw on the `ListView`. But we also run into a problem with getting the correct rectangle.

Perhaps, we need Windows help to solve this problem

**Comment by merriemcgaw** [comment]
I agree for the **ListView** that we would need to get Windows to fix it. I'm interested in how it renders on Win11, we should check that out. 

For the **CheckedListBox** I'm not 100% certain that we need to change the "marching ants" (dotted line) color at all given that the selection should be of a reasonable ratio for low vision users to be able to differentiate it from non-selected items. My suggestion is that we pop this question into the Accessibility Teams site and find out if it is a requirement. Are there situations where we are showing just the dotted line without the blue? Or that the blue selection could be on one item in the list and the dotted line could be somewhere else?

**Comment by SergeySmirnov-Akvelon** [comment]
Hi @merriemcgaw. I have investigated colors and found that orange color has low contrast for normal colors:
`Green` - 1.172
`Yellow` - 2.052
`Blue` (highlighted) - 1,874
`Red` - 1,825

I think we could replace it with alternating `Black` and `White`, as we did for the buttons in the `PropertyGrid`.
Firstly, both colors have a high contrast for `Blue` (`Black`- 4.667 and `White` - 4.498). 
Secondly, whatever color the user chooses as the background for the `CheckedListbox`, either `Black` or `White` will have the necessary contrast.

We could add a special method for the `CheckedListBox` to draw a `Black` and `White` dashed rectangle for the selected item. Thus, our changes will be locate and will not affect the behavior of other controls.
![image](https://user-images.githubusercontent.com/23376742/125596204-9252ae1e-4908-4b3e-857e-43db7db8e555.png)
@merriemcgaw, could you please share your opinion?

**Comment by merriemcgaw** [comment]
I like that idea! Sounds like a plan to me.

**Comment by Philip-Wang01** [comment]
Verified the function in the 6.0.100-rc.1.21377.28. 
For the **CheckedListBox**, the color contrast of the focus dot line on selection items is above 3:1. 
For the **ListView**, according to the above description, we still need to get Windows to fix it.
**CheckedListBox**:
![7281](https://user-images.githubusercontent.com/86937911/127295761-ba5dc824-3876-4830-8ae3-2f9dad7c6630.png)
**ListView**:
![72802](https://user-images.githubusercontent.com/86937911/127295772-ff53dd7a-4479-4813-927e-10878b44f300.png)
